executions/graphql-kotlin-dataloader/src/test/kotlin/com/expediagroup/graphql/dataloader/KotlinDataLoaderRegistryFactoryTest.kt

```diff
@@ -39,10 +47,63 @@ class KotlinDataLoaderRegistryFactoryTest {
     fun `generate registry with basic loader`() {
         val mockLoader: KotlinDataLoader<String, String> = object : KotlinDataLoader<String, String> {
             override val dataLoaderName: String = "MockDataLoader"
-            override fun getDataLoader(): DataLoader<String, String> = mockk()
+            override fun getDataLoader(graphQLContext: GraphQLContext?, graphQLContextMap: Map<*, Any>?): DataLoader<String, String> = mockk()
         }
 
         val registry = KotlinDataLoaderRegistryFactory(listOf(mockLoader)).generate()
         assertEquals(1, registry.dataLoaders.size)
     }
+
+    @Test
+    fun `generate registry with minimal compilable loader throws TODO`() {
+        val mockLoader: KotlinDataLoader<String, String> = object : KotlinDataLoader<String, String> {
+            override val dataLoaderName = "Unimplemented"
+        }
+        assertFailsWith(NotImplementedError::class) {
+            KotlinDataLoaderRegistryFactory(listOf(mockLoader)).generate()
+        }
+    }
+
+    @Test
+    fun `generate registry with GraphQLContext`() = runBlocking {
+        val mockLoader = object : KotlinDataLoader<String, String> {
+            override val dataLoaderName = "withGraphQLContext"
+            override fun getDataLoader(graphQLContext: GraphQLContext?, graphQLContextMap: Map<*, Any>?): DataLoader<String, String> {
+                val options = DataLoaderOptions.newOptions().setBatchLoaderContextProvider {
+                    graphQLContext
+                }
+                return DataLoaderFactory.newDataLoader({ keys, environment ->
+                    keys.map { (environment.getContext() as GraphQLContext).toString() }.toMono().toFuture()
+                }, options)
+            }
+        }
+        val context = object : GraphQLContext {
+            override fun toString(): String {
+                return "blah"
+            }
+        }
+        val registry = KotlinDataLoaderRegistryFactory(mockLoader).generate(context)
+        val result = registry.getDataLoader<String, String>(mockLoader.dataLoaderName).load("123")
+        registry.dispatchAll()
+        assertEquals(result.await(), "blah")
+    }
+
+    @Test
+    fun `generate registry with GraphQLContextMap`() = runBlocking {
+        val mockLoader = object : KotlinDataLoader<String, String> {
+            override val dataLoaderName = "withGraphQLContext"
+            override fun getDataLoader(graphQLContext: GraphQLContext?, graphQLContextMap: Map<*, Any>?): DataLoader<String, String> {
+                val options = DataLoaderOptions.newOptions().setBatchLoaderContextProvider {
+                    graphql.GraphQLContext.of(graphQLContextMap)
+                }
+                return DataLoaderFactory.newDataLoader({ keys, environment ->
```

Comment:
example of how code could use this to set the batch loader context to the GraphQLContext

Reply:
Hey @josephlbarnett first of all thank you for the PR, contributions are always welcomed!

i would like to mention a couple of thoughts with this approach:

1. As mentioned in the slack conversation, we kind of solved this problem in a different way using the `BatchLoaderEnvironment` keys context as it feels more natural and adds access not only to the `GraphQLContext` but also the whole `DataFetchingEnvironment`, at the end the `GraphQLContext` exists only once in memory  and we would be just working with references
2. With this approach we are kind of adding an implicit dependency between the DataLoaders and the `GraphQLContext` as the only option for a context in a `BatchLoaderEnvironment` will only be `GraphQLContext`, in any case i would check if we can work with generics instead of an explicit type

having pointed all these thoughts i just think that passing the `GraphQLContext` by key would just work fine and you can just leverage on the extension fuctions that we added to facilitate the access to the context

```kotlin
DataLoaderFactory.newDataLoader { requests, environment: BatchLoaderEnvironment ->
  val context = environment.getGraphQLContext()
  // do something with requests and context
 }
```

Reply:
FYI these extension functions were released on version 6.1.0

Reply:
Hey @samuelAndalon , thanks for your reply here.  My concern with the keys approach is less about performance/memory, and more about semantics of the batch contexts.  Looking at the graphql-java docs on this, my read is that the keyContexts really can/should be able to be different for each object loaded by the batch loader.  While I don't really have a use case for that, an approach that forces it to be the same for every loaded object when there is already an existing "overall context" object is (to me at least) a surprising/confusing one when reading the code later, or writing new dataloaders that need to remember to call the correctly overloaded .load() method.

I have updated the PR to be a little less tied to `GraphQLContext`, and instead depend on a `DataLoaderOptions` per request, which allows control of the options beyond just the context object, which I think is a good change.  Generating that options object (in a new `KotlinDataLoaderOptionsFactory` class) is still tied to `GraphQLContext` though so not sure it addresses your generic concern?  Could potentially also pass the GraphQLRequest through to generate the DataLoaderOptions, but not sure that's necessary?

Reply:
@josephlbarnett we had an internal discussion about this and we don't think its a good idea,
as mentioned we solved this problem by passing the `DataFetchingEnvironment` per each `load` call.

about your concern
> an approach that forces it to be the same for every loaded object 
this is partially incorrect, as `DataFetchingEnvironment` is unique per data fetcher, which happens to have a reference to the `GraphQLContext`.

Try the approach we recommend using the extension fuctions and passing the `dataFetchingEnvironment` to each `load` invokation

Reply:
ok thanks for the correction there, and thanks for having the discussion.  Your recommended approach definitely does work.

We'll probably still do something wrapping KotlinDataLoaderRegistryFactory to expose the graphql context to the DataLoaderOptions so that we only have to think about it when constructing the dataloaders and not in each resolver that calls load(), but respect your team's decision on this.